src/trunk.c

```diff
@@ -7984,40 +8059,35 @@ trunk_print_pivots(platform_log_handle *log_handle,
    for (uint16 pivot_no = 0; pivot_no < trunk_num_pivot_keys(spl, node);
         pivot_no++)
    {
-      char key_string[128];
-      trunk_key_to_string(
-         spl, trunk_get_pivot(spl, node, pivot_no), key_string);
+      key               pivot = trunk_get_pivot(spl, node, pivot_no);
       trunk_pivot_data *pdata = trunk_get_pivot_data(spl, node, pivot_no);
       if (pivot_no == trunk_num_pivot_keys(spl, node) - 1) {
          platform_log(log_handle,
                       "| %.*s | %12s | %12s | %11s | %9s | %5s | %5s |\n",
                       24,
-                      key_string,
+                      key_string(spl->cfg.data_cfg, pivot),
                       "",
                       "",
                       "",
                       "",
                       "",
                       "");
-         platform_log(log_handle,
-                      "| Full key: '%.*s'\n",
-                      (int)sizeof(key_string),
-                      key_string);
       } else {
          platform_log(log_handle,
                       "| %.*s | %12lu | %12lu | %11lu | %9lu | %5ld | %5lu |\n",
                       24,
-                      key_string,
+                      key_string(spl->cfg.data_cfg, pivot),
                       pdata->addr,
                       pdata->filter.addr,
                       pdata->num_tuples_whole + pdata->num_tuples_bundle,
                       pdata->num_kv_bytes_whole + pdata->num_kv_bytes_bundle,
                       pdata->srq_idx,
                       pdata->generation);
-         platform_log(log_handle,
-                      "| Full key: '%.*s'\n",
-                      (int)sizeof(key_string),
-                      key_string);
+      }
+      if (key_is_user_key(pivot)) {
```

Comment:
I plan to do an audit to convert all printing to use the `key_string` and `message_string` macros.  But I want to do it as a separate PR, since this one is already huge.

Also, I'm not sure what full key-printing function we have, if any.